typings/downshift.flow.js

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react'
+
+// @flow
+
+type A11StatusMessageOptions = {
+  highlightedIndex: number,
+  highlightedValue: any,
+  inputValue: string,
+  isOpen: boolean,
+  itemToString: (item: any) => string,
+  previousResultCount: number,
+  resultCount: number,
+  selectedItem: any,
+}
+
+type ChangeOptions = {
+  selectedItem: any,
+  previousItem: any,
+}
+
+type StateChangeOptions = {
+  highlightedIndex: number,
+  inputValue: string,
+  isOpen: boolean,
+  selectedItem: any,
+}
+
+type GetRootPropsOptions = {
+  refKey: string,
+}
+
+type GetItemPropsOptions = {
+  index: number,
+  item: any,
+}
+
+type ControllerStateAndHelpers = {
+  // prop getters
+  getRootProps: (options: GetRootPropsOptions) => any,
+  getButtonProps: (options: GetButtonPropsOptions) => any,
+  getLabelProps: (options: GetLabelPropsOptions) => any,
+  getInputProps: (options: GetInputPropsOptions) => any,
+  getItemProps: (options: GetItemPropsOptions) => any,
+
+  // actions
+  openMenu: () => void,
+  closeMenu: () => void,
+  toggleMenu: () => void,
+  selectItem: (item: any) => void,
+  selectItemAtIndex: (index: number) => void,
+  selectHighlightedItem: (index: number) => void,
+  setHighlightedItem: (index: number) => void,
+  clearSelection: () => void,
+
+  // state
+  highlightedIndex: number,
+  inputValue: string,
+  isOpen: boolean,
+  selectedItem: any,
+}
+
+type GetInputPropsOptions = React.InputHTMLAttributes
+
+type GetLabelPropsOptions = React.LabelHTMLAttributes
+
+type GetButtonPropsOptions = {
+  // actions
+  clearSelection: () => void,
+  closeMenu: () => void,
+  openMenu: () => void,
+  selectHighlightedItem: () => void,
+  selectItem: (item: any) => void,
+  selectItemAtIndex: (index: number) => void,
+  setHighlightedIndex: (index: number) => void,
+  toggleMenu: (state: boolean) => void,
+
+  // state
+  highlightedIndex: number,
+  inputValue: string,
+  isOpen: boolean,
+  selectedItem: any,
+}
+
+type ChildrenFunction = (options: ControllerStateAndHelpers) => React.Node<*>
+
+type downshiftProps = {
```

Comment:
`onOuterClick?: () => void` is missing

Reply:
Can you capitalize `downshiftProps` to match the other type definitions?

Reply:
```js
  onUserAction?: (
    options: StateChangeOptions,
    stateAndHelpers: ControllerStateAndHelpers,
  ) => void
``` is also missing

Reply:
`itemCount: number | void` should be added

Reply:
`id?: string` is missing

Reply:
`environment?: Environment` also needs to be added. You can get the definition for Environment from the TypeScript definition file